sklearn/cluster/_affinity_propagation.py

```diff
@@ -133,7 +141,13 @@ def affinity_propagation(S, preference=None, convergence_iter=15, max_iter=200,
                     if return_n_iter
                     else (np.array([0]), np.array([0] * n_samples)))
 
-    random_state = np.random.RandomState(0)
+    if random_state == 'warning':
+        random_state = 0
+        warnings.warn(("'random_state' has been introduced in 0.23. "
+                       "It could be manually set. "
+                       "Its value is defaulted to 0, will be 'None' "
+                       "from 0.25."), FutureWarning)
```

Comment:
Let me suggest e.g.

> 'random_state' has been introduced in 0.23. It will be set to None starting from 0.25 which means that results will differ at every function call. To silence this warning and obtain the same results as before 0.23, set it to 0.

Reply:
approved! :)